### PR TITLE
BAH-1628 | Care-Context definition change

### PR DESCRIPTION
**Pull request by SanoferSameera**

### CODE REVIEW

File: api/src/main/java/org/bahmni/module/hip/api/dao/impl/CareContextRepositoryImpl.java [hunk 1]
```diff
@@ -26,15 +26,20 @@ public List<PatientCareContext> getPatientCareContext(String patientUuid) {
...
+                "    case when care_context = 'PROGRAM' then program_name\n" +
+                "    else\n" +
+                "        concat (visit_type_name,\" / \",visit_startDate)\n" +
+                "    end as careContextName\n" +
                 "from\n" +
                 "    (\n" +
                 "        select\n" +
                 "            ppa.value_reference,p3.uuid,e.patient_id, p2.program_id, vt.visit_type_id , vt.name ,\n" +
+                "            v.date_started As visit_startDate,\n" +
+                "            concat(pn.given_name, ifnull(concat(' ',pn.middle_name),'')," +
```
Comment (SanoferSameera) [inline api/src/main/java/org/bahmni/module/hip/api/dao/impl/CareContextRepositoryImpl.java:40]: PreparedStatement is preferred

File: omod/src/main/java/org/bahmni/module/hip/web/controller/PrescriptionController.java [hunk 2]
```diff
@@ -35,30 +37,34 @@ public PrescriptionController(PrescriptionService prescriptionService, Validatio
...
+        if (visitStartDate == null || visitStartDate.isEmpty())
+            return ResponseEntity.badRequest().body(ClientError.noVisitTypeProvided());
         if (!validationService.isValidVisit(visitType))
             return ResponseEntity.badRequest().body(ClientError.invalidVisitType());
         if (!validationService.isValidPatient(patientId))
             return ResponseEntity.badRequest().body(ClientError.invalidPatientId());
-        List<PrescriptionBundle> prescriptionBundle =
-                prescriptionService.getPrescriptions(patientId, new DateRange(parseDate(fromDate), parseDate(toDate)), visitType);
+        List<PrescriptionBundle> prescriptionBundle = new ArrayList<>();
+        if(parseDate(visitStartDate).compareTo(parseDate(fromDate)) >= 0 && parseDate(visitStartDate).compareTo(parseDate(toDate)) < 0)
```
Comment (SanoferSameera) [inline omod/src/main/java/org/bahmni/module/hip/web/controller/PrescriptionController.java:58]: extract date comparision in dateUtil

File: omod/src/main/java/org/bahmni/module/hip/web/utils/DateUtils.java [hunk 3]
```diff
@@ -9,4 +9,9 @@ public static Date parseDate(String date) throws ParseException {
         SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd");
         return dateFormat.parse(date);
     }
+
```
Comment (SanoferSameera) [inline omod/src/main/java/org/bahmni/module/hip/web/utils/DateUtils.java:12]: check joda DateTime

File: api/src/main/java/org/bahmni/module/hip/api/dao/impl/ConsultationDaoImpl.java [hunk 4]
```diff
@@ -97,37 +96,33 @@ public List<Obs> getAllObs(String programName, Date fromDate, Date toDate, Patie
...
     @Override
-    public List<Order> getOrders(Patient patient, String visit, Date fromDate, Date toDate) {
-        List<Order> orders = orderService.getAllOrdersByPatient(patient);
-        return orders.stream().filter(order -> matchesVisitType(visit, order))
-                              .filter(order -> order.getEncounter().getVisit().getStartDatetime().after(fromDate))
-                              .filter(order -> order.getEncounter().getVisit().getStartDatetime().before(toDate))
-                              .filter(order -> order.getDateStopped() == null && !Objects.equals(order.getAction().toString(), ORDER_ACTION))
-                              .filter(order -> ORDER_TYPES.contains(order.getOrderType().getName()))
-                              .collect(Collectors.toList());
+    public List<Order> getOrders(Patient patient, String visit, Date visitStartDate) {
```
Comment (SanoferSameera) [inline api/src/main/java/org/bahmni/module/hip/api/dao/impl/ConsultationDaoImpl.java:120]: use visit object

Comment (SanoferSameera) [inline api/src/main/java/org/bahmni/module/hip/api/dao/impl/ConsultationDaoImpl.java:120]: use visit object

**Review by gsluthra — APPROVED** [review]

**Pull request merged**